### PR TITLE
Refactor shield satalites to use proximity_monitor making them more reliable

### PR DESCRIPTION
**Pull request by ItsTisme**
## About The Pull Request

Refactors the meteor satalites to use proximity_monitor making them MUCH more reliable

## Why It's Good For The Game

Previvously it would process and check if meteor is in range every 0.2 seconds via subsystem. This made it very possible that meteor flew past it before it even got detected, bypassing the shield. For it being station objective that feels rather silly. Especialy when meteors can actuely HIT THE SAT AND DESTROY IT

Also slight delay in toggleing on so you cant spam toggle it

## Changelog
:cl:
refactor: Nanotrasen has introducted new upgrades into the aging station shield statalites, they require a but longer to toggle on however
/:cl:

### CODE REVIEW

**Comment by Absolucy** [comment]
my downstream code where i did effectively the same thing may be useful: https://github.com/Monkestation/Monkestation2.0/tree/master/monkestation/code/modules/meteor_shield

i also implemented multi-z coverage, tracking which turfs are covered, and making debris drift towards the station.

also tbh i disagree with making them take time to toggle on as a "trade-off" for making them actually reliable - the reliability was not an intended feature, it was just because using process() to loop thru meteors was a horrible idea, bc meteors that make it thru would cause explosions, which would cause the sats to skip ticks, allowing more meteors through to cause more explosions, causing the sats to skip more process ticks, and so on.

File: code/modules/station_goals/meteor_shield.dm [hunk 1]
```diff
@@ -94,34 +95,42 @@
...
+		var/obj/effect/meteor/meteor_to_destroy = proximity_check_mob
 		if(space_los(meteor_to_destroy))
 			var/turf/beam_from = get_turf(src)
 			beam_from.Beam(get_turf(meteor_to_destroy), icon_state="sat_beam", time = 5)
 			if(meteor_to_destroy.shield_defense(src))
 				qdel(meteor_to_destroy)
 
 /obj/machinery/satellite/meteor_shield/toggle(user)
+	if(user)
+		src.balloon_alert(user, "you're looking for the [active ? "off" : "on"] button on the [src]")
```
Comment (Ghommie) [inline code/modules/station_goals/meteor_shield.dm:114]:
```suggestion
		balloon_alert(user, "looking for [active ? "off" : "on"] button")
```
Balloon messages short be fairly short.

File: code/modules/station_goals/meteor_shield.dm [hunk 2]
```diff
@@ -94,34 +95,42 @@
...
+	if(user && !do_after(user, 2 SECONDS, src, IGNORE_HELD_ITEM))
+		return FALSE;
```
Comment (Ghommie) [inline code/modules/station_goals/meteor_shield.dm:116]:
```suggestion
		return FALSE
```
DM oft doesn't need a statement separator on new lines outside of the precompiler iirc.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/station_goals/meteor_shield.dm [hunk 3]
```diff
@@ -94,34 +95,42 @@
...
-	if(!active)
-		return
-	for(var/obj/effect/meteor/meteor_to_destroy in GLOB.meteor_list)
-		if(meteor_to_destroy.z != z)
-			continue
-		if(get_dist(meteor_to_destroy, src) > kill_range)
-			continue
+/obj/machinery/satellite/meteor_shield/Initialize(mapload)
+	. = ..()
+	proximity_monitor = new(src,0)
```
Comment (ZephyrTFA) [inline code/modules/station_goals/meteor_shield.dm:100]: 0... what? use a named param here please

File: code/modules/station_goals/meteor_shield.dm [hunk 4]
```diff
@@ -94,34 +95,42 @@
...
+
+/obj/machinery/satellite/meteor_shield/HasProximity(atom/movable/proximity_check_mob)
+	. = ..()
+	if(istype(proximity_check_mob, /obj/effect/meteor))
```
Comment (ZephyrTFA) [inline code/modules/station_goals/meteor_shield.dm:104]: this should be inverted to be an early return

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/modules/station_goals/meteor_shield.dm (hunk 3 above)
Comment (Ghommie) [reply code/modules/station_goals/meteor_shield.dm:100]:
```suggestion
	proximity_monitor = new(src, /* range = */ 0)
```

**Pull request merged**